app/uk/gov/hmrc/helptosave/controllers/HelpToSaveAuth.scala

```diff
@@ -46,8 +48,17 @@ class HelpToSaveAuth(htsAuthConnector: AuthConnector) extends BaseController wit
   override def authConnector: AuthConnector = htsAuthConnector
 
   private type HtsAction = Request[AnyContent] ⇒ Future[Result]
+  private type HtsActionTyped[A] = Request[AnyContent] ⇒ A ⇒ Future[Result]
```

Comment:
perhaps `HtsActionWithRetrieval` is more descriptive? If you have this you don't really need `HtsActionWithNino` since it's can be written in terms of this type

Reply:
ok thank you